### DISCUSSION TITLE
Recommended way to handle API path like `projects/{project}/tasks`

### DISCUSSION THREAD

**Discussion by edalzell** [JustSteveKing/laravel-transporter#46] [Q&A]
I don't see any examples in the docs, what is the recommended way to deal w/ an API with a variable in it (see title)?

Right now I'm doing something like this, but it feels wrong:
<img width="1019" alt="CleanShot 2023-04-25 at 13 40 28@2x" src="https://user-images.githubusercontent.com/6069653/234398498-ee674ea1-c88d-481f-b25c-6c69014e3966.png">

**Comment by JustSteveKing** [comment]
Hey @edalzell yeah this is something that bothered me too - it is definitely something I would like to solve as well!

Happy to start a discussion on this to see what can be achieved for the next release?

**Reply by edalzell** [reply]
Sure, might be nice to have something like `setPathParam('project', $project)` and that will end up substituted in a path that has `{project}` in it? Kinda like route/model binding?

Easier but maybe not as nice is a `replaceInPath('{project}', $project)`?

**Reply by JustSteveKing** [reply]
I definitely think I should test a few different approaches and see what work/feels the best
